spec/models/supplemental_file_spec.rb

```diff
@@ -15,11 +15,51 @@
 require 'rails_helper'
 
 describe SupplementalFile do
+  let(:subject) { FactoryBot.create(:supplemental_file) }
+
+  describe 'validations' do
+    describe 'file type' do
+      context 'non-caption file' do
+        let(:subject) { FactoryBot.create(:supplemental_file, :with_attached_file) }
+        it 'should skip validation' do
+          expect(subject.valid?).to be_truthy
+        end
+      end
+      context 'VTT/SRT caption file' do
+        let(:subject) { FactoryBot.create(:supplemental_file, :with_caption_file, :with_caption_tag) }
+        it 'should validate' do
+          expect(subject.valid?).to be_truthy
+        end
+      end
+      context 'non-VTT/non-SRT caption file' do
+        let(:subject) { FactoryBot.create(:supplemental_file, :with_attached_file) }
+        it 'should not validate' do
+          subject.tags = ['caption']
+          subject.save
+          expect(subject.valid?).to be_falsey
```

Comment:
I think this could be simplified if the subject is built but not created:
```suggestion
        let(:subject) { FactoryBot.build(:supplemental_file, :with_attached_file, :with_caption_tag) }
        it 'should not validate' do
          expect(subject.valid?).to be_falsey
```